server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -384,6 +388,17 @@ private Version computeMinIndexCompatVersion() {
         return Version.min(this, fromId(bwcMajor * 1000000 + bwcMinor * 10000 + 99));
     }
 
+    private Version computePreviousFirstMinor() {
+        List<Version> allVersions = DeclaredVersionsHolder.DECLARED_VERSIONS;
+        for (int i = allVersions.size() - 1; i >= 0; i--) {
+            Version v = allVersions.get(i);
+            if (v.before(this) && (v.minor < this.minor || v.major < this.major)) {
+                return fromId(v.major * 1000000 + v.minor * 10000 + 99);
+            }
+        }
+        throw new IllegalArgumentException("couldn't find any released versions of the minor before [" + this + "]");
```

Comment:
can't we do the same that we do for the previousMajor? Why go through all the declared version?

Reply:
PreviousMajor calculation is simpler, I think, you always only have two mayors and only need to calculate the former.
For "first" previous minor, you would e.g. want 8.1.2 -> 8.0.0, 8.0.1 -> 7.17.0, 7.15.4 -> 17.14.0.0
This gets a bit tricky expecially around the major version changes, i.e. getting the previous minor for all 8.0.x releases. Which minor do I pick then?
Potentially we could detect that special case, but I don't see a good way of getting the last minor of the previous major without looking at all the released versions and pick the "last" one (with revision 0 btw.)

Reply:
I see, ignore me :)